### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by utsargo**
### Description
- Refactor `main.rb` with required changes
- Create an `options.file` which will contain all the ui options
- Create `Options` class in the options.rb file

### Team

- @utsargo 
- @JacintaNgwu

### CODE REVIEW

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @utsargo and @JacintaNgwu, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- The app works correctly ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- There are no changes in the app.rb file that let me use inline comments, but please 🙏🏽  there are some lines of code that you need to refactor for the correct use of [SOLID](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) principles 👇🏽 

image

image

image

- [ ] All requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input, please 🙏🏽 refactor your code for the best practices of SOLID principles.

image

image

- [ ] These lines should've been placed in the `create_person()` method, please 🙏🏽 refactor your code for the best practices of SOLID principles. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by utsargo** [comment]
Hi @franclobo ,
Thanks for your review. In the project requirement, there is nothing required regarding `app.rb` file. Every instruction is explicitly for `main.rb` file. I've got your point but I think the changes you required should go under optional suggestions.

File: options.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#         '1 - List all books',
+#         '2 - List all people',
+#         '3 - Create a person',
+#         '4 - Create a book',
+#         '5 - Create a rental',
+#         '6 - List all rentals for a given person id',
+#         '7 - Exit'
+#         print options
+#     end of print_options
+# end of Options class
```
Comment (TedLivist) [inline options.rb:17]: - [ ] **[OPTIONAL]** Kindly remove these commented lines of code.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @utsargo and @JacintaNgwu,

✔️ Check for the existence of book before creating rental

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**